equinox-shipment/Domain/Container.fs

```diff
@@ -0,0 +1,56 @@
+module Container
+
+type Container = { id: string; shipmentIds: string[]; finalized: bool }
+
+module Events =
+
+    let [<Literal>] CategoryId = "Shipment"
+
+    let (|ForClientId|) (clientId: string) = FsCodec.StreamName.create CategoryId clientId
+
+    type Event =
+        | ContainerCreated    of containerId  : string
+        | ContainerFinalized  of shipmentIds  : string[]
+        | Snapshotted         of Container
+        interface TypeShape.UnionContract.IUnionContract
+
+    let codec = FsCodec.NewtonsoftJson.Codec.Create<Event>()
+
+module Fold =
+
+    type State = Container
+
+    let initial: State = { id = null; shipmentIds = [||]; finalized = false }
+
+    let evolve (state: State) (event: Events.Event): State =
+        match event with
+        | Events.Snapshotted        snapshot    -> snapshot
+        | Events.ContainerCreated   containerId -> { state with id = containerId }
+        | Events.ContainerFinalized shipmentIds -> { state with shipmentIds = shipmentIds }
+
+
+    let fold: State -> Events.Event seq -> State =
+        Seq.fold evolve
+
+    let isOrigin (event: Events.Event) =
+        match event with
+        | Events.Snapshotted _ -> true
+        | _ -> false
+
+    let snapshot state = Events.Snapshotted state
+
+type Command =
+    | Create   of containerId : string
+    | Finalize of shipmentIds : string[]
+
+let interpret (command: Command) (state: Fold.State): Events.Event list =
+    match command with
+    | Create containerId  ->
+        [ Events.ContainerCreated containerId ]
```

Comment:
- even initialization events should be handled idempotently

I know this is kinda a placeholder event representing something real in the Actual Domain
but containerId is not really useful, and it also avoids you using null on L24 (in general I prefer to have `type State = ContainerState option` and model the State as a State Machine)

Reply:
Modified thanks!